### ISSUE TITLE
[FEATURE REQUEST] Use USB-C port for flash drives

### ISSUE THREAD

**Issue by me25542** [prusa3d/Prusa-Firmware-Buddy#4370]
### Printer model

xBuddy based; MK4(s), MK3.9(s), MK3.5(s)

### Describe the feature

Using the USB-C port on the xBuddy board for USb flash drives would allow mainly for the USB drive to not be exposed (unsightly, and vulnerable to being hit accidently, broken, etc.) when the primary means of printing is remote via PrusaConnect or PrusaLink. I don't know for certain if this is possible with the hardware on the xBuddy board (to use the port in `Host` mode), but after looking through the schematics I think it *might*.

### Expected functionality

The printer would be able to use USB flash drives inserted into the USB-C port for printing, etc.

**Comment by neofir3** [comment]
Yes, that would be great if that were possible. Then the data wouldn't always have to go through the xLCD cable to the front to the USB port. This might also make the (B)GCode upload faster. At the time, I actually assumed that you could also plug a USB stick into the back. But unfortunately, so far that's only been possible in front of the display. Although an xBuddy board with an NVMe slot would also be a dream. :)

**Comment by danopernis** [comment]
@me25542 not high on the priority list, but looking briefly at the schematics, it might be possible. Looks like a nice hacking project for anybody interested :slightly_smiling_face: 

@neofir3 Buddy, XBuddy and XLBuddy boards are already shipped to thousands of customers so they are not likely to contain any new hardware. I can't disclose details about next generation of the motherboard.

**Comment by Tupson444** [comment]
Good idea. If it is possible, It can be used as a backup if the front port breaks (I don't know how likely that is to happen, but anyway), or if you only have USB-C type flash at hand.

If it is not limited by hardware, then simply why not :)

**Comment by me25542** [comment]
Still want this issue open.

**Comment by me25542** [comment]
Still would like this at some point, assuming it is possible.

**Comment by me25542** [comment]
Still would like this.

**Comment by me25542** [comment]
This issues appears to be stale; if anyone else still wants it open please comment, otherwise I will close it.

**Comment by drfish** [comment]
I too would love to hook up a small thumbdrive to the side of the xbuddy instead of protruding from the front of the printer where it might get bumped.

**Comment by Fred255255** [comment]
Might also help with the printer freezing (no error displayed) issue???

**Comment by me25542** [comment]
This issue should stay open.

**Comment by me25542** [comment]
This issue should stay open.

**Comment by me25542** [comment]
This issue should stay open.
